BedrockCore.cpp

```diff
@@ -135,6 +135,15 @@ BedrockCore::RESULT BedrockCore::peekCommand(unique_ptr<BedrockCommand>& command
             if (!_db.beginTransaction(exclusive ? SQLite::TRANSACTION_TYPE::EXCLUSIVE : SQLite::TRANSACTION_TYPE::SHARED)) {
                 STHROW("501 Failed to begin " + (exclusive ? "exclusive"s : "shared"s) + " transaction");
             }
+            if (exclusive) {
+                const uint64_t remainingTimeUS = _getRemainingTime(command, false);
+                if ((BedrockCommand::DEFAULT_BLOCKING_TRANSACTION_COMMIT_LOCK_TIMEOUT * 1000ull) < remainingTimeUS) {
+                    command->setTimeout(BedrockCommand::DEFAULT_BLOCKING_TRANSACTION_COMMIT_LOCK_TIMEOUT);
+                    const int64_t newRemainingTimeUS = _getRemainingTime(command, false);
+                    _db.setTimeout(newRemainingTimeUS);
+                    SINFO("Reduced timeout because of blocking commit lock from " << STIMESTAMP(STimeNow() + remainingTimeUS) << " to " << STIMESTAMP(STimeNow() + newRemainingTimeUS));
+                }
+            }
```

Comment:
Is there a particular command (or set of commands) we're targeting with this? I wonder if this catches al lthe most relevant cases. It should catch the most common "query is too slow" case.

We may want to do it here as well: https://github.com/Expensify/Bedrock/blob/aa77afd175e72ec65f86754d2ef5263dd1d9c084/BedrockCore.cpp#L229

Reply:
oh, I missed that, I guess we I could move the new code to a function we can call it from this place too.

Reply:
I tested sending a request in `AddComment.peek` and you were right that in such case the timeout was not reduced in blocking commit.

I refactored the code into a function `void reduceCommandTimeout(unique_ptr<BedrockCommand>& command, uint64_t timeoutMS);` and now I'm calling it from `peekCommand` and `processCommand`